backend/dataall/modules/loader.py

```diff
@@ -29,49 +31,187 @@ class ModuleInterface(ABC):
     An interface of the module. The implementation should be part of __init__.py of the module
     Contains an API that will be called from core part
     """
-    @classmethod
+    @staticmethod
     @abstractmethod
-    def is_supported(cls, modes: List[ImportMode]):
-        pass
+    def is_supported(modes: List[ImportMode]) -> bool:
+        """
+        Return True if the module interface supports any of the ImportMode and should be loaded
+        """
+        raise NotImplementedError("is_supported is not implemented")
+
+    @classmethod
+    def name(cls) -> str:
+        """
+        Returns name of the module. Should be the same if it's specified in the config file
+        """
+        return _remove_module_prefix(cls.__module__)
+
+    @staticmethod
+    def depends_on() -> List[Type['ModuleInterface']]:
+        """
+        It describes on what modules this ModuleInterface depends on.
+        It will be used to eventually load these module dependencies. Even if a dependency module is not active
+        in the config file.
+
+        The default value is no dependencies
+        """
+        return []
 
```

Comment:
We are changing `is_supported` and `depends_on` from @classmethods to @staticmethods because they do not reference anything about the class itself right? Or am I missing something else?

Reply:
Maybe we can discuss it today as well, but could we have something more like.... I don't understand why we need to repeat the is_supported in every module
```
class MLStudioApiModuleInterface(ModuleInterface):
    """Implements ModuleInterface for MLStudio GraphQl lambda"""

    def __init__(self):
        import dataall.modules.mlstudio.api
        self.import_mode = ImportMode.API
        log.info("API of sagemaker mlstudio has been imported")

```
And then implement the `is_supported` in the loader?
```
class ModuleInterface(ABC):
    """
    An interface of the module. The implementation should be part of __init__.py of the module
    Contains an API that will be called from core part
    """
    @classmethod
    def is_supported(cls, modes: List[ImportMode]):
        if cls.import_mode in modes:
```

Reply:
Unfortunately, if we enter the constructor of `ModuleInterface` to set modes we will also load `import dataall.modules.mlstudio.api` and we want ot avoid it if the mode is not supported